packages/lib/modules/pool/actions/unstake/useClaimAndUnstakeSteps.tsx

```diff
@@ -28,7 +34,14 @@ export function useClaimAndUnstakeSteps(unstakeParams: UnstakeParams): {
   )
 
   const steps = useMemo((): TransactionStep[] => {
-    const steps = [relayerApprovalStep, claimAndUnstakeStep]
+    const steps = [claimAndUnstakeStep]
+
+    if (shouldSignRelayerApproval) {
+      steps.unshift(signRelayerStep)
```

Comment:
doesn't all those unshift look a little difficult to understand?

Reply:
some steps are conditional so doing it like above isn't the best way to build the array

i can do it first to last if last to first is too difficult?

```
    const steps: TransactionStep[] = []

    if (hasUnclaimedBalRewards) {
      steps.push(minterApprovalStep)
    }

    steps.push(shouldSignRelayerApproval ? signRelayerStep : relayerApprovalStep)
    steps.push(claimAllRewardsStep)

    return steps
```